packages/world/src/utils/ForceFieldUtils.sol

```diff
@@ -90,12 +91,10 @@ library ForceFieldUtils {
    */
   function removeFragment(EntityId forceField, EntityId fragment) internal {
     require(!fragment._getProgram().exists(), "Can't remove a fragment with a program");
-
-    (EnergyData memory machineData,) = updateMachineEnergy(forceField);
-    Energy._setDrainRate(
-      forceField, machineData.drainRate - MACHINE_ENERGY_DRAIN_RATE - Fragment._getExtraDrainRate(fragment)
-    );
-    Fragment._deleteRecord(fragment);
+    require(Fragment._getExtraDrainRate(fragment) == 0, "Fragment must not have an extra drain rate");
```

Comment:
so with this you cant expand into fragments with sleeping players?

Reply:
Actually now that we are preventing mining the forcefield, we could still allow this, but might be safer to still prevent it unless we think it is important to support